molecule/default/tasks/openshift_route.yml

```diff
@@ -0,0 +1,219 @@
+    service: hello-kubernetes
+    namespace: default
+    name: test1
+    wait: yes
+  register: route
+
+- name: Assert that the condition is properly set
+  assert:
+    that:
+      - route.duration is defined
+      - route.result.status.ingress.0.conditions.0.type == 'Admitted'
+      - route.result.status.ingress.0.conditions.0.status == 'True'
+
+- name: Attempt to hit http URL
+  uri:
+    url: 'http://{{ route.result.spec.host }}'
+    return_content: yes
+  register: result
+
+- name: Assert the page content is as expected
+  assert:
+    that:
+      - not result.redirected
+      - result.status == 200
+      - result.content == 'Hello OpenShift!\n'
+
+- name: Create edge-terminated route that allows insecure traffic
+  community.okd.openshift_route:
+    service: hello-kubernetes
+    namespace: default
+    name: hello-kubernetes-https
+    tls:
+      insecure_policy: allow
+    termination: edge
+  register: route
+
+- name: Attempt to hit http URL
+  uri:
+    url: 'http://{{ route.result.spec.host }}'
+    return_content: yes
+  until: result is successful
+  retries: 10
+  register: result
+
+- name: Assert the page content is as expected
+  assert:
+    that:
+      - not result.redirected
+      - result.status == 200
+      - result.content == 'Hello OpenShift!\n'
+
+- name: Attempt to hit https URL
+  uri:
+    url: 'https://{{ route.result.spec.host }}'
+    validate_certs: no
+    return_content: yes
+  until: result is successful
+  retries: 10
+  register: result
+
+- name: Assert the page content is as expected
+  assert:
+    that:
+      - not result.redirected
+      - result.status == 200
+      - result.content == 'Hello OpenShift!\n'
+
+- name: Alter edge-terminated route to redirect insecure traffic
+  community.okd.openshift_route:
+    service: hello-kubernetes
+    namespace: default
+    name: hello-kubernetes-https
+    tls:
+      insecure_policy: redirect
+    termination: edge
+  register: route
+
+- name: Attempt to hit http URL
+  uri:
+    url: 'http://{{ route.result.spec.host }}'
+    return_content: yes
+    validate_certs: no
+  until:
+    - result is successful
+    - result.redirected
+  retries: 10
+  register: result
+
+- name: Assert the page content is as expected
+  assert:
+    that:
+      - result.redirected
+      - result.status == 200
+      - result.content == 'Hello OpenShift!\n'
+
+- name: Attempt to hit https URL
+  uri:
+    url: 'https://{{ route.result.spec.host }}'
+    validate_certs: no
+    return_content: yes
+  until: result is successful
+  retries: 10
+  register: result
+
+- name: Assert the page content is as expected
+  assert:
+    that:
+      - not result.redirected
+      - result.status == 200
+      - result.content == 'Hello OpenShift!\n'
+
+- name: Alter edge-terminated route disallow
+  community.okd.openshift_route:
+    service: hello-kubernetes
+    namespace: default
+    name: hello-kubernetes-https
+    tls:
+      insecure_policy: null
```

Comment:
This line is causing the integration tests to fail with a  "msg" of "value of insecure_policy must be one of: allow, redirect, got: None found in tls".

Reply:
odd that it passed locally, must have been a stale version of the collection or something. Added an explicit `disable` option for disallowing insecure traffic, should be good to go now